tests/system/framework/misc/ssh.py

```diff
@@ -0,0 +1,175 @@
+from __future__ import annotations
+
+import shlex
+from functools import wraps
+from time import sleep
+from typing import Any, Callable, ParamSpec
+
+from pytest_mh.conn import Connection, Process, ProcessError, ProcessLogLevel, ProcessResult
+
+from . import to_list_of_strings
+
+
+class SSHKillableProcess(object):
+    """
+    Run an asynchronous process that requires ``SIGTERM`` to be terminated.
+    """
+
+    def __init__(
+        self,
+        client: Connection,
+        argv: list[Any],
+        *,
+        cwd: str | None = None,
+        env: dict[str, Any] | None = None,
+        input: str | None = None,
+        log_level: ProcessLogLevel = ProcessLogLevel.Full,
+    ) -> None:
+        """
+        :param client: SSH client.
+        :type client: Connection
+        :param argv: Command to run.
+        :type argv: list[Any]
+        :param cwd: Working directory, defaults to None (= do not change)
+        :type cwd: str | None, optional
+        :param env: Additional environment variables, defaults to None
+        :type env: dict[str, Any] | None, optional
+        :param input: Content of standard input, defaults to None
+        :type input: str | None, optional
+        :param log_level: Log level, defaults to ProcessLogLevel.Full
+        :type log_level: ProcessLogLevel, optional
+        """
+        if env is None:
+            env = {}
+
+        argv = to_list_of_strings(argv)
+        command = shlex.join(argv)
+        pidfile = "/tmp/.mh.sshkillableprocess.pid"
```

Comment:
The standard location for PID files is $runstatedir, that is, <$prefix/var/run/program-name/>.

(The .../program-name/ directory is only necessary when the program stores more than one file.  If there's only the PID file, you can put it directly under the main `$prefix/var/run/` dir.)

See also:

-  <https://www.gnu.org/prep/standards/html_node/Directory-Variables.html#Directory-Variables>
-  <https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch05s13.html>
-  <https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch03s15.html>

Reply:
Although for a test suite there might be reasons to not follow the standard...

Reply:
@pbrezina your input here is much welcome

Reply:
This NGINX Unit discussion also contains useful info: <https://github.com/nginx/unit/pull/1235>

Reply:
As Pavel mentioned this class is not needed, so I removed it